src/components/pages/coach/CoachDashboard/index.jsx

```diff
@@ -24,11 +38,508 @@ import { ReactComponent as MorningImg } from  '../../../../assets/icons/morning.
+                this._fetchAllUsers(customers);
+            } else if(response.status === 401 || response.status === 403){
+                // Forbidden (token invalid)
+                this._loginUser();
+            } else {
+                this.setState({
+                    syncing: false
+                }, () => console.log("Error",response));
+            }
+        }).catch((error) => {
+            this.setState({
+                syncing: false
+            }, () => console.log("Error",error));
+        });
+    }
+
+    _loginUser = () => {
+        // Start syncing animation
+        if(!this.state.syncing){
+            this.setState({
+                syncing: true
+            });
+        }
+
+        // Login to get JWT
+        axios.post("https://charmnode.herokuapp.com/login", {
+            "username": process.env.REACT_APP_NODE_USER,
+            "password": process.env.REACT_APP_NODE_PASS
+        })
+        .then((response) => {
+            if(response.status === 200){
+                this.setState({
+                    jwt: response.data.token
+                }, () => this._fetchNodeUsers(response.data.token));
+            } else {
+                this.setState({
+                    syncing: false
+                }, () => console.log("Error",response));
+            }
+        }, (error) => {
+            this.setState({
+                syncing: false
+            }, () => console.log("Error",error));
+        });
+    }
+
+    _fetchUsersWithoutCache = () => {
+        clientNoCache.query({
+            query: GET_DATA,
+            variables: { "token": localStorage.getItem("fprint") }
+        }).then(({data}) => {
+            console.log("Fetched data directly from server.");
+            if(data.userSelf){
+                this.setState({
+                    coachusers: {
+                        userdata: data.userSelf
+                    }
+                });
+            }
+        })
+        .catch(error => {
+            this.setState({
+                syncing: false
+            }, () => console.log("Error",error));
+        })
+    }
+
+    _fetchUsersFirstTime = () => {
+        clientOnlyCache.query({
+            query: GET_DATA,
+            variables: { "token": localStorage.getItem("fprint") }
+        }).then(({data}) => {
+            console.log("Fetched data from cache.");
+            if(data.userSelf){
+                this.setState({
+                    coachusers: {
+                        userdata: data.userSelf
+                    }
+                }, () => this._fetchUsersWithoutCache());
+            }
+        })
+        .catch(error => {
+            this.setState({
+                syncing: false
+            }, () => console.log("Error",error));
+        })
+    }
+    
+    _fetchAllUsers = (customers) => {
+        clientNoCache.query({
+            query: GET_USERS_ALL,
+            variables: { "token": localStorage.getItem("fprint") }
```

Comment:
"fprint" doesn't seem to be the best naming for an JWT in local storage.
I would prefer to name it anonymousJWT or something along those lines,
cause it's the JWT of the anonymous user.

Reply:
I have to agree on that one. They are a remenant of darker times. 🌋